### PR TITLE
Add SendText and SendData to DataChannel

### PR DESCRIPTION
**Pull request by DylanJ**
I found myself needing text messages over a data channel and so I've added the functionality here.

This PR adds the method `SendText([]byte)`

I am totally open to any changes to get this merged as I think it is quite useful.

Thanks for your time!

### CODE REVIEW

File: datachannel.go [hunk 1]
```diff
@@ -73,11 +73,22 @@ func NewDataChannel(o unsafe.Pointer) *DataChannel {
 	return c
 }
 
-func (c *DataChannel) Send(data []byte) {
+// Send a message over a datachannel in either text or binary mode.
+func (c *DataChannel) Send(data []byte, binary bool) {
```
Comment (arlolra) [inline datachannel.go]: I'd rather not change the signature of this method.  It'd be fine to add `SendText()` and have both call an unexported `sendInternal()` with a bool.

**Comment by DylanJ** [comment]
@arlolra I've addressed your comments and reverted my change to `Send()`. There is now a `SendText()` for sending text messages :)

File: datachannel.go [hunk 2]
```diff
@@ -76,11 +76,21 @@ func NewDataChannel(o unsafe.Pointer) *DataChannel {
 	return c
 }
 
+// Send a message over a DataChannel in binary mode.
 func (c *DataChannel) Send(data []byte) {
+	c.sendInternal(data, true)
+}
+
+// SendText sends a message over the DataChannel in text mode.
+func (c *DataChannel) SendText(data []byte) {
```
Comment (arlolra) [inline datachannel.go]: I would think the signature would be `SendText(data string)` and then call `sendInternal([]byte(data), false)`.

Comment (DylanJ) [inline datachannel.go]: You're right. I've made the change.

File: demo/chat/chat.go [hunk 3]
```diff
@@ -149,7 +150,7 @@ func endChat() {
 func sendChat(msg string) {
 	line := username + ": " + msg
 	fmt.Println("[sent]")
-	dc.Send([]byte(line))
+	dc.SendText([]byte(line))
```
Comment (arlolra) [inline demo/chat/chat.go]: `dc.SendText(line)`

**Comment by arlolra** [comment]
Thanks @DylanJ!  I'll try to get to your other pull soon.

**Pull request merged**